AK/URL.cpp

```diff
@@ -402,7 +402,7 @@ bool URL::code_point_is_in_percent_encode_set(u32 code_point, URL::PercentEncode
     case URL::PercentEncodeSet::Component:
         return code_point_is_in_percent_encode_set(code_point, URL::PercentEncodeSet::Userinfo) || "$%&+,"sv.contains(code_point);
     case URL::PercentEncodeSet::ApplicationXWWWFormUrlencoded:
-        return code_point >= 0x7E || !(is_ascii_alphanumeric(code_point) || "!'()~"sv.contains(code_point));
+        return code_point >= 0x7E || !(is_ascii_alphanumeric(code_point) || "*-._"sv.contains(code_point));
```

Comment:
Given this:

> The application/x-www-form-urlencoded percent-encode set is the [component percent-encode set](https://url.spec.whatwg.org/#component-percent-encode-set) and U+0021 (!), U+0027 (') to U+0029 RIGHT PARENTHESIS, inclusive, and U+007E (~).

Wouldn't this be easier to understand if implemented the same way as `URL::PercentEncodeSet::Component`?

i.e. `return code_point_is_in_percent_encode_set(code_point, URL::PercentEncodeSet::Component) || "!'()~"sv.contains(code_point);`